### PR TITLE
Bug 1417709 - Add -moz-appearance: -moz-mac-vibrant-titlebar-light/dark and system colors with the same name.

### PR DESCRIPTION
**Pull request by mstange**
These are the servo changes that need to go with [bug 1417709](https://bugzilla.mozilla.org/show_bug.cgi?id=1417709).

r? emilio

---

- [ ] `./mach build -d` does not report any errors
- [ ] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [ ] There are tests for these changes OR
- [x] These changes do not require tests because without these changes Firefox doesn't build.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19255)

### CODE REVIEW

File: components/script/parser.out [hunk 1]
```diff
@@ -0,0 +1,8079 @@
+Created by PLY version 3.3 (http://www.dabeaz.com/ply)
```
Comment (jdm) [inline components/script/parser.out]: This looks like an accident.

Comment (mstange) [reply components/script/parser.out]: Thanks for catching that, yes, it was an accident. I've fixed it.

**Comment by jdm** [comment]
```
error[E0243]: wrong number of type arguments: expected 1, found 0
  --> components/style/gecko_bindings/sugar/ns_com_ptr.rs:26:6
   |
26 | impl nsCOMPtr {
   |      ^^^^^^^^ expected 1 type argument

error: aborting due to previous error

error: Could not compile `style`.
```

**Comment by mstange** [comment]
Hmm, not sure why that happened. I grabbed the newly-generated bindings from https://treeherder.mozilla.org/#/jobs?repo=try&revision=79aa11b6616009280fb7d3e224e6e6af1258adb8&selectedJob=146539533 which includes #19313 which was the latest change to `structs.rs`.

**Comment by jdm** [comment]
Right, and that probably added a type parameter to the nsCOMPtr type that ns_com_ptr.rs modifies.

**Comment by emilio** [comment]
Not exactly. That happens when the `structs.rs` are from a debug build, not from a release build. nsCOMPtr uses different template parameters on debug for code-size reasons (https://bugzilla.mozilla.org/show_bug.cgi?id=1363754)

**Comment by mstange** [comment]
Oops, thanks Emilio. Looks like I accidentally clicked on the wrong build on treeherder.

**Pull request merged**